### ISSUE TITLE
[Feature Request]: keyboard navigation 

### ISSUE THREAD

**Issue by halr9000** [anapnoe/stable-diffusion-webui-ux#205]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What would your feature do ?

As a keyboard jockey, I would like for application consistent keystrokes for navigating a complex interface
- some thoughtful attention to tab order, cursor keys would be good
- ESC key should always exit a subordinate view/dialog
- perhaps when panel open, pressing numbers will go to ordinal tab, eg "2" opens TI

As a frequent user of Loras, and other favorites, I want a keystroke to navigate directly to that panel
- eg alt-L -> open Lora, alt-Q for quick settings, T for themes and so on

As a frequent tweaker, I want focus and navigate the side panels using only the keyboard
- but I also want to quickly get back to image creation, so a universal hide/switch key is important
- something easy which isn't needed otherwise, like backtick or tilde?
- when hiding panels, I would always reset focus back to the positive prompt, if the last used main tab is t2i or i2i. makes it quicker to generate w/o grabbing the mouse

### Proposed workflow

1. load UX, main tab should be t2i, and keyboard focus should be positive prompt so that I can type prompts without touching the mouse
2. alt-L to choose lora, "/" to focus search field, start typing, typeahead search finds what I want quickly, pressing enter inserts text into prompt
3. ESC hides current panel, continue typing prompt
4. ctrl-enter to generate
5. realize I want to change ckpt, click ~, cursor right twice, down once, ENTER to select active model. Selection is done, no need to have the screen open, so just exit it (saving me having to press ~ or ESC)
6. ctrl-enter will go ahead and queue image for when the model is finished loading

**Comment by anapnoe** [comment]
Shortcuts are a must to do but I have to think how to make use of them to the workspaces ui-ux as well 
in workspaces you can have i2i and t2i on the same view and each custom view can have an instance of a component that is already been used in another workspace as well 
I need to do some more thinking on this but it will come after workspaces is released
![workspaces_custom](https://github.com/anapnoe/stable-diffusion-webui-ux/assets/124302297/92b70cac-a866-4d3f-be35-ad2e0dc765c2)

**Comment by halr9000** [comment]
> Shortcuts are a must to do but I have to think how to make use of them to the workspaces ui-ux as well in workspaces you can have i2i and t2i on the same view and each custom view can have an instance of a component that is already been used in another workspace as well I need to do some more thinking on this but it will come after workspaces is released ![workspaces_custom](https://user-images.githubusercontent.com/124302297/284691122-92b70cac-a866-4d3f-be35-ad2e0dc765c2.png)
> 
> if you want you can write down a list with the shortcuts keys may be dynamically set from a shortcuts panel something like
> 
> 1. Focus txt2img prompt  -> default key
>    ...

Oh sweet that's looking great. Good to know shortcuts are on your mind. I'll close this out for now then. ESC key is an easy one though, it just does the one thing. :D
